src/Packetizer/ThinPacketizer.c

```diff
@@ -0,0 +1,127 @@
+/**@file
+ * This file is part of the TASTE Runtime Common.
+ *
+ * @copyright 2022 N7 Space Sp. z o.o.
+ *
+ * TASTE Runtime Common was developed under a programme of,
+ * and funded by, the European Space Agency (the "ESA").
+ *
+ * Licensed under the ESA Public License (ESA-PL) Permissive,
+ * Version 2.3 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://essr.esa.int/license/list
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Packetizer.h"
+
+#include <assert.h>
+#include <string.h>
+
+#include "PacketizerInternal.h"
+#include "SpacePacketInternal.h"
+
+#define THIN_PACKET_PRIMARY_HEADER_SIZE 4u
+
+void
+Packetizer_init(Packetizer* const self)
+{
+}
+
+size_t
+Packetizer_packetize(Packetizer* const self,
+                     const Packetizer_PacketType packetType,
+                     const uint16_t source,
+                     const uint16_t destination,
+                     uint8_t* const packetPointer,
+                     const size_t dataOffset,
+                     const size_t dataSize)
+{
+    // Unused in this implementation
+    (void)self;
+    (void)packetType;
+    (void)source;
+    (void)dataOffset;
+
+    assert(destination <= SPACE_PACKET_MAX_APID);
+    assert(packetPointer != NULL);
+    assert(dataOffset == THIN_PACKET_PRIMARY_HEADER_SIZE);
+    assert(dataSize < SPACE_PACKET_MAX_PACKET_DATA_SIZE);
```

Comment:
what about `assert(dataSize > 0)`? I see `dataSize - 1` later in the code

Reply:
I forgot to copy that assert from default Packetizer. 
Fixed.